### ISSUE TITLE
ID_FS_USAGE etc are not set in udev immediately after "stratis fs create" 

### ISSUE THREAD

**Issue by mvollmer** [stratis-storage/stratisd#2162]
After creating a new filesystem with "stratis fs create" or via the D-Bus API, the ID_FS_USAGE etc properties are not set in udev.

```
$ stratis fs create pool0 fsys
$ udevadm info $(readlink /stratis/pool0/fsys)
P: /devices/virtual/block/dm-7
N: dm-7
L: 0
S: disk/by-id/dm-uuid-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
S: mapper/stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
S: disk/by-id/dm-name-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: DEVPATH=/devices/virtual/block/dm-7
E: DEVNAME=/dev/dm-7
E: DEVTYPE=disk
E: MAJOR=253
E: MINOR=7
E: SUBSYSTEM=block
E: USEC_INITIALIZED=198498677
E: DM_UDEV_PRIMARY_SOURCE_FLAG=1
E: DM_ACTIVATION=1
E: DM_NAME=stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: DM_UUID=stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: DM_SUSPENDED=0
E: DM_UDEV_RULES_VSN=2
E: E2_SIZE=0
E: DEVLINKS=/dev/disk/by-id/dm-uuid-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4 /dev/mapper/stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4 /dev/disk/by-id/dm-name-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: TAGS=:systemd:
```

The properties how up after `udevadm trigger`, or a reboot.

```
$ udvadm trigger
$ udevadm info $(readlink /stratis/pool0/fsys)
P: /devices/virtual/block/dm-7
N: dm-7
L: 0
S: disk/by-uuid/f4529e57-96b2-4c2b-a8c6-9f64bab89ae4
S: mapper/stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
S: disk/by-id/dm-name-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
S: disk/by-id/dm-uuid-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: DEVPATH=/devices/virtual/block/dm-7
E: DEVNAME=/dev/dm-7
E: DEVTYPE=disk
E: MAJOR=253
E: MINOR=7
E: SUBSYSTEM=block
E: USEC_INITIALIZED=198498677
E: DM_UDEV_PRIMARY_SOURCE_FLAG=1
E: DM_UDEV_RULES_VSN=2
E: DM_NAME=stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: DM_UUID=stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: DM_SUSPENDED=0
E: ID_FS_UUID=f4529e57-96b2-4c2b-a8c6-9f64bab89ae4
E: ID_FS_UUID_ENC=f4529e57-96b2-4c2b-a8c6-9f64bab89ae4
E: ID_FS_TYPE=xfs
E: ID_FS_USAGE=filesystem
E: E2_SIZE=0
E: DEVLINKS=/dev/disk/by-uuid/f4529e57-96b2-4c2b-a8c6-9f64bab89ae4 /dev/mapper/stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4 /dev/disk/by-id/dm-name-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4 /dev/disk/by-id/dm-uuid-stratis-1-a1654aed04c14ec69e595aa151b32b76-thin-fs-f4529e5796b24c2ba8c69f64bab89ae4
E: TAGS=:systemd:
```

Cockpit wont show a block device as a filesystem unless it has `ID_FS_USAGE==filesystem`.

**Comment by jbaublitz** [comment]
@mvollmer I noticed you're still using the `/stratis` path. Are you using the most recent released version of stratisd or are you using the code in #2146? If you are using the code in #2146, that now uses the path `/dev/stratis` and requires a udev rule to be present on the system to appropriately create the symlink. This udev rule is located in the stratisd repo under the `udev` directory. My guess is that those attributes will show up if you have the appropriate udev rule in the system folder (`/usr/lib/udev/rules.d`) and create a pool and filesystem on the #2146 branch. Can you give that a try?

**Comment by mulkieran** [comment]
@mvollmer Some of our tests require being run from an rpm of the HEAD of the development branch for stratisd. Likely we can give you an rpm that distributes all the files into the correct places if you're interested in following this up more right now. Let us know.

**Comment by mvollmer** [comment]
> Are you using the most recent released version of stratisd or are you using the code in #2146?

I was trying to use stratisd-2.0.0-2.fc32.x86_64, but maybe I messed it up while playing with #2146. I never noticed the missing ID_FS_USAGE before, so that's a strong indication that I created this issue for myself somehow.

But I just did `dnf reinstall stratisd` and the problem persists.

I did copy the udev rule and the stratis_uuid_to_names helper while playing with #2146 since otherwise I wouldn't get any symlinks in /dev/stratis, and I was running stratisd directly from the Git checkout. I noticed this issue at that point, and then also tried with stratisd from the installed RPM, with the same results.
